cvmfs/cache_ram.cc

```diff
@@ -0,0 +1,250 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+#include "cvmfs_config.h"
+#include "cache_ram.h"
+
+#include <errno.h>
+#include <assert.h>
+#include <new>
+#include <string.h>
+
+#include "kvstore.h"
+#include "smalloc.h"
+#include "util/posix.h"
+#include "util_concurrency.h"
+
+using namespace std;  // NOLINT
+
+namespace cache {
+
+
+int RamCacheManager::AddFd(const ReadOnlyFd &fd) {
```

Comment:
We should probably have an upper, enforced limit for the number of open file descriptors.